flow360/component/simulation/entity_info.py

```diff
@@ -0,0 +1,142 @@
+"""Desearlizer for entity info retrieved from asset metadata pipeline."""
+
+from abc import ABCMeta, abstractmethod
+from typing import List, Literal, Union
+
+import pydantic as pd
+
+from flow360.component.simulation.framework.entity_registry import EntityRegistry
+from flow360.component.simulation.primitives import Edge, GenericVolume, Surface
+
+
+class EntityInfoModel(pd.BaseModel, metaclass=ABCMeta):
+    """Base model for asset entity info JSON"""
+
+    model_config = pd.ConfigDict(
+        ##:: Pydantic kwargs
+        extra="ignore",
+        frozen=False,
+        populate_by_name=True,
+        validate_assignment=True,
+        validate_default=True,
+    )
+
+    @abstractmethod
+    def get_full_list_of_boundaries(self, attribute_name: str = None) -> list:
+        """
+        Helper function.
+        Get the full list of boundary names. If it is geometry then use supplied attribute name to get the list.
+        """
+
+
+class GeometryEntityInfo(EntityInfoModel):
+    """Data model for geometry entityInfo.json"""
+
+    type_name: Literal["GeometryEntityInfo"] = pd.Field("GeometryEntityInfo", frozen=True)
+    face_ids: list[str] = pd.Field(
+        [],
+        description="A full list of faceIDs/model IDs that appear in the geometry.",
+        alias="faceIDs",
+    )
+    face_attribute_names: List[str] = pd.Field(
+        [],
+        description="A full list of attribute names that the user can"
+        "select to achieve grouping of faces. It has same length as `grouped_faces`",
+        alias="faceAttributeNames",
+    )
+    grouped_faces: List[List[Surface]] = pd.Field(
+        [[]],
+        description="The resulting list "
+        "of `Surface` entities after grouping using the attribute name.",
+        alias="groupedFaces",
+    )
```

Comment:
instead of list of lists, can we have better structure? Maybe:
```python
grouped_faces: List[FaceGroup]

class FaceGroup:
    name: str
    items: List[Surface]
```
so we can access this group by name

By the way, we mix all, Face, Surface, Boundary now, for example: `grouped_faces: List[List[Surface]]`. Is is necessary?

Reply:
I prefer separating the tags from their grouping:
1. Front end/ python client can directly know what tag keys are available.
2. Even if we put the name under FaceGroup like this we will need to loop through this list to get the index so there is no improvement in terms of usability.